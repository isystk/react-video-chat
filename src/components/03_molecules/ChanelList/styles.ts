import { css, keyframes } from '@emotion/css'

const useStyles = (theme) => css`{
  chanelList: {
    overflowY: 'scroll',
    backgroundColor: '#444444',
    '& .tag': {
      width: '100%',
      padding: '10px',
      display: 'flex',
      cursor: 'pointer',
      position: 'relative',
      '&.active': {
        backgroundColor: '#666666',
      },
      '& .head': {
        width: '34px',
        height: '34px',
        borderRadius: '50%',
        overflow: 'hidden',
      },
      '& .head img': {
        width: '100%',
      },
      '& .text': {
        width: '90%',
        marginLeft: '10px',
        display: 'block',
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
      },
      '& .text .name': {
        width: '90%',
        color: '#fff',
        fontSize: '16px',
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
      },
      '& .text .dec': {
        width: '100%',
        color: '#888888',
        fontSize: '13px',
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
      },
      '& .num': {
        width: '20px',
        height: '20px',
        lineHeight: '20px',
        borderRadius: '50%',
        backgroundColor: 'rgb(10, 156, 200)',
        color: '#eee',
        fontSize: '8px',
        textAlign: 'center',
        position: 'absolute',
        right: '5%',
        top: '30%',
      },
    },
    '@media screen and (min-width:992px)': {
      height: '100%',
    },
  },
}`

export { useStyles }
