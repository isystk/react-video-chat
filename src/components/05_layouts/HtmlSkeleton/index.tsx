import React, { Children, FC, useState } from 'react'
import { connect } from '@/components/hoc'
import { ContainerProps, WithChildren } from 'types'
import useAppRoot from '@/stores/useAppRoot'
import Header from '@/components/04_organisms/Header'
import SideMenu from '@/components/04_organisms/SideMenu'
import Title, { TitleProps } from './Title'
import Head from 'next/head'
import { isReactElement } from '@/utils/general/object'
import { CssBaseline } from '@material-ui/core'

/** HtmlSkeleton Props */
export type HtmlSkeletonProps = WithChildren
/** Presenter Props */
export type PresenterProps = HtmlSkeletonProps & {
  title?: TitleProps['children']
  description: string
}

/** Presenter Component */
const HtmlSkeletonPresenter: FC<PresenterProps> = ({
  main,
  title,
  description,
  isMenuOpen,
  setMenuOpen,
  children,
  ...props
}) => (
  <>
    <Head>
      {/* タイトル */}
      <title>{title}</title>
      {/* favicon */}
      <link rel="icon" href="/favicon.ico" />
      {/* ブラウザテーマカラー */}
      <meta name="theme-color" content="#ffffff" />
      {/* サイト概要 */}
      <meta name="description" content={description} />
      {/* OGP 画像URL */}
      <meta property="og:image" content={'/ogp-image.png'} />
      {/* OGP タイトル */}
      <meta name="og:title" content={title} />
      {/* OGP サイト概要 */}
      <meta name="og:description" content={description} />
      {/* OGP Twitterカード */}
      <meta name="twitter:card" content="summary" />
      {/* apple ポータブル端末 アイコン */}
      <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
      {/* manifest.json */}
      <link rel="manifest" href="/manifest.json" />
    </Head>
    <Context.Provider value={main}>
      <CssBaseline />
      <Header isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
      <div>{children}</div>
      <SideMenu isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
    </Context.Provider>
  </>
)

/** Container Component */
const HtmlSkeletonContainer: React.FC<
  ContainerProps<HtmlSkeletonProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  let title: TitleProps['children'] | undefined = undefined
  const main = useAppRoot()
  const [isMenuOpen, setMenuOpen] = useState(false)

  if (!main) return <></>

  const APP_NAME = process.env.APP_NAME

  children = Children.map(children, (child) =>
    isReactElement(child) && child.type === Title
      ? (title = `${child.props.children} | ${APP_NAME}`) && undefined
      : React.cloneElement(child)
  )
  if (!title) title = process.env.APP_NAME

  const description = `${APP_NAME} is an application that allows you to casually chat with your friends!`
  return presenter({
    main,
    title,
    description,
    isMenuOpen,
    setMenuOpen,
    children,
    ...props,
  })
}

export const Context = React.createContext(null)

export default connect<HtmlSkeletonProps, PresenterProps>(
  'HtmlSkeleton',
  HtmlSkeletonPresenter,
  HtmlSkeletonContainer
)

// Sub Component
export type { TitleProps }
export { Title }
