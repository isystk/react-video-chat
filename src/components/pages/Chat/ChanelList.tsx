import * as React from 'react'
import Main from '@/services/main'
import * as _ from 'lodash'

// ↓ 表示用のデータ型
interface IProps {
  rtcClient: Main
}

const ChanelList = ({ rtcClient }: IProps) => {
  return (
    <div id="myChatList" className="chat_list">

      {_.map(rtcClient.chanels, (chanel, index) => (
        <div className="chatListTag" key={index}>
          <div className="head">
            <img src={chanel.photo} alt="" />
          </div>
          <div className="mytext">
            <div className="name">{chanel.name}</div>
            <div className="dec">{chanel.detail}</div>
          </div>
          <div className="msg_num">7</div>
        </div>
      ))}
    </div>
  )
}

export default ChanelList
