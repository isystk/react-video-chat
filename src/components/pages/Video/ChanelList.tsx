import * as React from 'react'
import Main from '@/services/main'
import { Button, TextField } from '@material-ui/core'
import { useCallback, useEffect, useState } from 'react'

// ↓ 表示用のデータ型
interface IProps {
  rtcClient: Main
}

const ChanelList = ({ rtcClient }: IProps) => {
  const [windowHeight, setWindowHeight] = useState(0)

  useEffect(() => {
    setWindowHeight(window.innerHeight)
  }, [])

  return (
    <div id="myChatList" className="chat_list" style={appStyle(windowHeight)}>
      <div className="chatListTag">
        <div className="head">
          <img src="images/friends/Alpha_Team.png" alt="" />
        </div>
        <div className="mytext">
          <div className="name">Alpha_Team</div>
          <div className="dec">Alpha チーム</div>
        </div>
        <div className="msg_num">3</div>
      </div>

      <div className="chatListTag active">
        <div className="head">
          <img src="images/friends/Beta_Team.png" alt="" />
        </div>
        <div className="mytext">
          <div className="name">Beta_Team</div>
          <div className="dec">Beta チーム</div>
        </div>
        <div className="msg_num">2</div>
      </div>

      <div className="chatListTag">
        <div className="head">
          <img src="images/friends/BigBoss.png" alt="" />
        </div>
        <div className="mytext">
          <div className="name">Big Boss</div>
          <div className="dec">管理者</div>
        </div>
      </div>

      <div className="chatListTag">
        <div className="head">
          <img src="images/friends/David.png" alt="" />
        </div>
        <div className="mytext">
          <div className="name">David</div>
          <div className="dec">エンジニア</div>
        </div>
        <div className="msg_num">7</div>
      </div>
    </div>
  )
}

const appStyle = (vh) => {
  return {
    height: vh - 64,
    overflowY: 'scroll',
  }
}

export default ChanelList
