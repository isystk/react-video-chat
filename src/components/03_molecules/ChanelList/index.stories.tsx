import { Meta, Story } from '@storybook/react'
import React from 'react'
import ChanelList from './index'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'
import ChanelService from '@/services/chanel'

export default {
  title: '03_molecules/ChanelList',
  component: ChanelList,
} as Meta

const Template: Story = (props) => {
  const main = new MainService(() => ({}))
  main.setName('isystk')
  main.setRoomId('test')
  main.addChanel(
    new ChanelService(
      main,
      'all',
      'すべて',
      'all',
      'images/friends/Alpha_Team.png',
      'ルーム内のすべてのメンバー'
    )
  )
  main.addChanel(
    new ChanelService(
      main,
      'own',
      '自分',
      'all',
      'images/friends/BigBoss.png',
      'BigBoss'
    )
  )
  main.setChanelId('all')

  return (
    <Context.Provider value={main}>
      <ChanelList />
    </Context.Provider>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
