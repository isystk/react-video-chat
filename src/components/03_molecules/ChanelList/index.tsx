import * as React from 'react'
import * as _ from 'lodash'
import { ContainerProps, WithChildren } from 'types'
import * as styles from './styles'
import { FC, useContext } from 'react'
import { connect } from '@/components/hoc'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'

/** ChanelListProps Props */
export type ChanelListProps = WithChildren
/** Presenter Props */
export type PresenterProps = ChanelListProps & {
  main
}

/** Presenter Component */
const ChanelListPresenter: FC<PresenterProps> = ({ main, ...props }) => (
  <>
    <div className={styles.chanelList}>
      {_.map(main.chanels, (chanel, index) => (
        <div
          className={`tag ${main.selectChanelId === chanel.id ? 'active' : ''}`}
          key={index}
          onClick={async () => await main.setChanelId(chanel.id)}
        >
          <div className="head">
            <img src={chanel.photo} alt="" />
          </div>
          <div className="text">
            <div className="name">{chanel.name}</div>
            <div className="dec">{chanel.detail}</div>
          </div>
          {0 < chanel.chat.getUnreadCount() && (
            <div className="num">{chanel.chat.getUnreadCount()}</div>
          )}
        </div>
      ))}
    </div>
  </>
)

/** Container Component */
const ChanelListContainer: React.FC<
  ContainerProps<ChanelListProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const main = useContext<MainService | null>(Context)
  if (!main) return <></>

  return presenter({
    children,
    main,
    ...props,
  })
}

export default connect<ChanelListProps, PresenterProps>(
  'ChanelList',
  ChanelListPresenter,
  ChanelListContainer
)
