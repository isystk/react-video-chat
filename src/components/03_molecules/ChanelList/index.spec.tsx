import React, { useState } from 'react'
import renderer from 'react-test-renderer'
import ChanelList from './index'
import '@testing-library/jest-dom/extend-expect'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'
import ChanelService from '@/services/chanel'

describe('ChanelList', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))
    main.setName('isystk')
    main.setRoomId('test')
    main.addChanel(
      new ChanelService(
        main,
        'all',
        'すべて',
        'all',
        'images/friends/Alpha_Team.png',
        'ルーム内のすべてのメンバー'
      )
    )
    main.addChanel(
      new ChanelService(
        main,
        'own',
        '自分',
        'all',
        'images/friends/BigBoss.png',
        'BigBoss'
      )
    )
    main.setChanelId('all')

    const component = renderer.create(
      <Context.Provider value={main}>
        <ChanelList />
      </Context.Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
