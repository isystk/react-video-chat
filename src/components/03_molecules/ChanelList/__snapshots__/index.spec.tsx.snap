// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChanelList Match Snapshot 1`] = `
<div
  className="css-l9oj8i"
>
  <div
    className="tag active"
    onClick={[Function]}
  >
    <div
      className="head"
    >
      <img
        alt=""
        src="images/friends/Alpha_Team.png"
      />
    </div>
    <div
      className="text"
    >
      <div
        className="name"
      >
        すべて
      </div>
      <div
        className="dec"
      >
        ルーム内のすべてのメンバー
      </div>
    </div>
  </div>
  <div
    className="tag "
    onClick={[Function]}
  >
    <div
      className="head"
    >
      <img
        alt=""
        src="images/friends/BigBoss.png"
      />
    </div>
    <div
      className="text"
    >
      <div
        className="name"
      >
        自分
      </div>
      <div
        className="dec"
      >
        BigBoss
      </div>
    </div>
  </div>
</div>
`;
